framework/Samplers/LimitSurfaceSearch.py

```diff
@@ -463,8 +466,12 @@ def localStillReady(self,ready): #,lastOutput=None
     testError = np.sum(np.abs(np.subtract(coarseGridTestMatix,coarseGridOldTestMatix)))
     if len(testMatrixDict) > 0:
       # compute the error
-      testError += np.sum(np.abs(np.subtract(testMatrixDict,oldTestMatrixDict)))
-    if (testError > self.errorTolerance):
+      sizeGrid = float(coarseGridTestMatix.size)
+      testError += np.sum(np.abs(np.subtract(testMatrixDict,oldTestMatrixDict)))/(sizeGrid+float(testMatrixDict.size))
+    else:
+      testError/= float(coarseGridTestMatix.size)
+
+    if (testError > self.errorTolerance) or newSizeLsFunctionValue == oldSizeLsFunctionValue:
```

Comment:
Could you explain the reason that you add the extra check here? i.e.
```
 newSizeLsFunctionValue == oldSizeLsFunctionValue
```

Reply:
It is because when the LS is preseeded by a prefilled data object, the localStillRrady is called at the initialization of the step for a number of times equal to the batchsize. If no check is present, if the persistence <= batchsize the run stop not even starting the iteration (see the issue 437)